src/pages/submit/linked-wearables.tsx

```diff
@@ -207,19 +244,48 @@ export default function SubmitLinkedWearables() {
     })
   }
 
+  const handleErrorOption = (
+    input: string,
+    fieldKey: string,
+    field: ListSectionType['section'],
+    validator: ListSectionValidator
+  ) => {
+    const error = validator(input)
+    if (input !== '' && error) {
+      setListSectionErrors({
+        ...listSectionErrors,
+        [field]: [...listSectionErrors[field], fieldKey],
+      })
+      editor.error({ [field]: state.error[field] || error })
+    } else {
+      setListSectionErrors({
+        ...listSectionErrors,
+        [field]: listSectionErrors[field].filter((key) => key !== fieldKey),
+      })
+      editor.error({ [field]: undefined })
+    }
+  }
+
   const handleProgrammaticallyGeneratedOption = () => {
     editor.set({
       ...state.value,
       programmatically_generated: !state.value.programmatically_generated,
     })
   }
 
-  const getListSection = (params: ListSectionType) => {
+  const getListSection = (
```

Comment:
What do you think about extracting this to a `ListSection` component? Maybe it's not necessary, but it looks like it has it's own logic we could encapsulate

Reply:
I tried it but it has a lot of logic linked to the state of the form (for example the functions handleAdd, handleEdit, etc...) and it didn't seem right to me neither to pass by argument those functions nor to pass the complete state of the form to the component so that it defines those functions internally